docs/specifications/tests/DNSSEC-TP/dnssec16.md

```diff
@@ -0,0 +1,220 @@
+          server IP to the *Mixed Delete CDS* set.
+       2. Else, add the name server IP address to the *Delete CDS*
+          set.
+       3. Go to next name server IP.
+    3. Get the DNSKEY and its RRSIG records, if any, from the
+       *DNSKEY RRsets* for the same name server IP.
+    4. If there are no DNSKEY records, then do:
+       1. Add name server IP address to the *No DNSKEY RRset* set.
+       2. Go to next name server IP.
+    5. Repeat the following steps for each CDS record:
+       1. Compare the key tag from the CDS record with the calculated
+          key tags for the DNSKEY records.
+       2. If the CDS record does not match any DNSKEY record then add
+          the name server IP address and key tag to the
+          *No Match CDS With DNSKEY* set.
+       3. Else, if there is no RRSIG for the DNSKEY RRset created by
+          the DNSKEY record that the CDS record points at then add the
+          name server IP address and key tag of CDS record to the
+          *DNSKEY Not Signed By CDS* set.
+    6. If there are no RRSIG records for the CDS RRset, then add the
+       name server IP address to the *CDS Not Signed* set.
+    7. Else, for each RRSIG (CDS) do:
+       1. If the key tag of the RRSIG does not match any DNSKEY then
+          add the name server IP address and key tag to the
+          *CDS Signed By Unknown DNSKEY* set.
+       2. Else, if the RRSIG cannot be validated by the DNSKEY it
+          refers to by key tag, then add the name server IP and RRSIG
+          key tag to the *CDS Invalid RRSIG* set.
+    8. Go to next name server IP address.
+
+8.  If the *No DNSKEY RRset* set is non-empty, then output
+    *[DS16_CDS_WITHOUT_DNSKEY]* with all name server IP addresses
+    in the set.
+
+9.  If the *Mixed Delete CDS* set is
+    non-empty, then output *[DS16_MIXED_DELETE_CDS]* with all
+    name server IP addresses in the set.
+
+10. If the *Delete CDS* set is non-empty, then output
+    *[DS16_DELETE_CDS]* with all name server IP addresses.
+
+11. If the *No Match CDS With DNSKEY* set is non-empty then for each
+    key tag in the set output *[DS16_CDS_MATCHES_NO_DNSKEY]* with the
+    CDS key tag and the name server IP addresses in the set per key
+    tag.
+
+12. If the *DNSKEY Not Signed By CDS* set is non-empty then for
+    each key tag in the set output *[DS16_DNSKEY_NOT_SIGNED_BY_CDS]*
+    with the RRSIG key tag and the name server IP addresses in the set
+    per key tag.
+
+13. If the *CDS Not Signed* set is non-empty then output 
+    *[DS16_CDS_UNSIGNED]* with all name server IP addresses in the set.
+
+14. If the *CDS Signed By Unknown DNSKEY* set is non-empty then output
+    *[DS16_CDS_SIGNED_BY_UNKNOWN_DNSKEY]* with the name server IP
+    addresses in the set.
+
+15. If the *CDS Invalid RRSIG* set is non-empty then for each key tag
+    in the set output *[DS16_CDS_INVALID_RRSIG]* with the RRSIG key
+    tag and the name server IP addresses in the set per key tag.
```

Comment:
Have you considered moving this up to just after step 12? That way steps with the same shape come just after each other.